addons/test_mail/tests/test_ir_actions.py

```diff
@@ -3,19 +3,53 @@
 
 from odoo.addons.base.tests.test_ir_actions import TestServerActionsBase
 from odoo.addons.test_mail.tests.common import TestMailCommon
+from odoo.tests import tagged
+from odoo.tools import mute_logger
 
 
+@tagged('ir_actions')
 class TestServerActionsEmail(TestMailCommon, TestServerActionsBase):
 
+    def setUp(self):
+        super(TestServerActionsEmail, self).setUp()
+        self.template = self._create_template(
+            'res.partner',
+            {'email_from': '{{ object.user_id.email_formatted or object.company_id.email_formatted or user.email_formatted }}',
+             'partner_to': '%s' % self.test_partner.id,
+            }
+        )
+
+    @mute_logger('odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
     def test_action_email(self):
-        email_template = self._create_template('res.partner', {'partner_to': '%s' % self.test_partner.id})
-        self.action.write({'state': 'email', 'template_id': email_template.id})
-        self.action.with_context(self.context).run()
+        # initial state
+        self.assertEqual(len(self.test_partner.message_ids), 1,
+                         'Contains Contact created message')
+        self.assertFalse(self.test_partner.message_partner_ids)
+
+        # update action: send an email
+        self.action.write({
+            'mail_post_method': 'email',
+            'state': 'mail_post',
+            'template_id': self.template.id,
+        })
+        self.assertFalse(self.action.mail_post_autofollow, 'Email action does not support autofollow')
+
+        with self.mock_mail_app():
+            self.action.with_context(self.context).run()
+
         # check an email is waiting for sending
         mail = self.env['mail.mail'].sudo().search([('subject', '=', 'About TestingPartner')])
         self.assertEqual(len(mail), 1)
-        # check email content
+        self.assertTrue(mail.auto_delete)
         self.assertEqual(mail.body, '<p>Hello TestingPartner</p>')
+        self.assertFalse(mail.is_notification)
+        with self.mock_mail_gateway(mail_unlink_sent=True):
+            mail.send()
+
+        # no archive (message)
```

Comment:
Ok I think I understand the "keep archive" thing now, not sure from an end-user POV, but it doesn't matter too much as this is a super advanced feature.